### PR TITLE
remove based on version

### PR DESCRIPTION
**Pull request by njr-11**
Experiments with considering the version on delete(entity) and deleteAll(iterable of entities)

### CODE REVIEW

**Comment by njr-11** [comment]
#build

File: dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java [hunk 1]
```diff
@@ -570,6 +575,98 @@ public void testEmbeddedId() {
...
+                Order o1updated = orders.findById(o1id).orElseThrow();
+                o1updated.total = 11.99f;
+                return orders.save(o1updated);
+            }).get(30, TimeUnit.SECONDS);
+
+            try {
+                orders.delete(o1);
+                fail("Deletion must be rejected when the version doesn't match.");
+            } catch (DataException x) {
+                System.out.println("Deletion was rejected as it ought to be be when the version does not match.");
```
Comment (mswatosh) [inline dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java]: There's an extra 'be'

**Review by mswatosh — DISMISSED** [review]

File: dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java (hunk 1 above)
Comment (njr-11) [reply dev/io.openliberty.data.internal_fat_jpa/test-applications/DataJPATestApp/src/test/jakarta/data/jpa/web/DataJPATestServlet.java]: Thanks for spotting that, it's fixed under https://github.com/OpenLiberty/open-liberty/pull/25336/commits/7feef3d6707c156c0c33212287a656a643f358a4 along with a resolution to the TODO for the IdClass/version combo.

**Review by mswatosh — APPROVED** [review]

**Comment by njr-11** [comment]
#build
#libby

**Pull request merged**